### PR TITLE
WIP v3.0.0 of all packages - with tree-shaking

### PR DESCRIPTION
**Pull request by sdalonzo**

### CODE REVIEW

**Comment by unscsprt** [comment]
[Tree-shaking](https://trello.com/c/n6r9l61r/3-tree-shaking)

File: packages/core/src/Avatar.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React from 'react'
```
Comment (jes708) [inline packages/core/src/Avatar.js]: I think new core components should be added to v2 as well, so we can have backwards compatibility in v3.

File: packages/core/src/Breadcrumbs.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react'
```
Comment (jes708) [inline packages/core/src/Breadcrumbs.js]: Add to v2

File: packages/core/src/Avatar.js (hunk 1 above)
Comment (James300) [reply packages/core/src/Avatar.js]: You could look at it as incentive to move to v3 😜

Comment (jes708) [reply packages/core/src/Avatar.js]: I'm looking forward to inheriting the performance wins in v3 for sure. I still strongly recommend putting these components in v2. When working in a host app that consumes several dependencies that all need to switch over from v2 to v3, backwards compatibility really helps facilitate that transition.

File: packages/core/src/Banner.js [hunk 3]
```diff
@@ -2,48 +2,57 @@ import React from 'react'
...
+  applyVariations,
+  hasPaletteColor,
+  deprecatedColorValue,
+  deprecatedPropType
+} from './utils'
 
 const bannerColors = {
   green: {
     backgroundColor: 'green',
     color: 'white',
```
Comment (achoboter) [inline packages/core/src/Banner.js]: should we remove the old color use?

File: packages/core/src/Radio.js [hunk 4]
```diff
@@ -45,21 +46,31 @@ const RadioInput = styled.input`
...
-
   const borderAdjustedSize = size + 4
 
+  const radioIcon = getRadioIcon({ checked, size: borderAdjustedSize })
+
   return (
     <RadioWrap color={props.color} checked={checked} disabled={disabled}>
       <RadioInput type="radio" {...props} />
-      <RadioIcon name={radioIconName} size={borderAdjustedSize} />
+      {radioIcon}
```
Comment (achoboter) [inline packages/core/src/Radio.js]: `getRadioIcon` could just be a component

File: packages/core/src/Banner.js [hunk 5]
```diff
@@ -56,11 +65,16 @@ const Banner = props => {
     bannerColors[
       !props.bg && props.color === 'green' ? props.color : props.bg
     ] || {}
-  const icon = props.iconName || bannerColor.icon
+  const Icon = props.icon || bannerColor.icon
   const color =
     hasPaletteColor(props) || !bannerColor.color
       ? props.color
       : bannerColor.color
+  const header = React.isValidElement(props.header) ? (
```
Comment (achoboter) [inline packages/core/src/Banner.js]: I feel like this should first check if the `prop.header` exists, then check if it is a valid element or if we should create one, else header can be null.

Then we can remove the check on line 90 `{props.header && header}` and just use `{header}`, and then we aren't creating an empty header when `props.header` in undefined.

File: packages/core/test/__snapshots__/FormField.js.snap [hunk 6]
```diff
@@ -18,15 +18,15 @@ exports[`FormField renders 1`] = `
   color: inherit;
   background-color: transparent;
   border-radius: 2px;
-  border-width: 0px;
+  border-width: 1px;
   border-style: solid;
+  border-color: #c0cad5;
```
Comment (achoboter) [inline packages/core/test/__snapshots__/FormField.js.snap]: I don't know if this is an issue, but in this snap there are a lot of classes that have `border-color` twice

File: packages/core/test/__snapshots__/IconField.js.snap [hunk 7]
```diff
@@ -28,15 +29,15 @@ exports[`IconField renders 1`] = `
   color: inherit;
   background-color: transparent;
   border-radius: 2px;
-  border-width: 0px;
+  border-width: 1px;
   border-style: solid;
+  border-color: #c0cad5;
```
Comment (achoboter) [inline packages/core/test/__snapshots__/IconField.js.snap]: this snap file also has `border-color` twice a few times

File: packages/core/test/__snapshots__/Input.js.snap [hunk 8]
```diff
@@ -11,15 +11,15 @@ exports[`Input it renders 1`] = `
   color: inherit;
   background-color: transparent;
   border-radius: 2px;
-  border-width: 0px;
+  border-width: 1px;
   border-style: solid;
+  border-color: #c0cad5;
```
Comment (achoboter) [inline packages/core/test/__snapshots__/Input.js.snap]: this one too has `border-color` twice a few times

File: packages/core/test/__snapshots__/Select.js.snap [hunk 9]
```diff
@@ -42,6 +40,7 @@ exports[`Select renders 1`] = `
   border-width: 1px;
   border-style: solid;
   border-color: #c0cad5;
+  border-color: #007aff;
```
Comment (achoboter) [inline packages/core/test/__snapshots__/Select.js.snap]: double `border-color` with different values?

File: packages/icons/test/index.js [hunk 10]
```diff
@@ -13,20 +16,13 @@ test.each(iconList)('renders %s', (key, Component) => {
...
-    expect(json).toEqual(expected)
+  test('defaults title to name if not provided', () => {
+    const testRenderer = TestRenderer.create(<Icon name="Accessible" />)
+    const testInstance = testRenderer.root
+    expect(testInstance.children[0].props.title).toBe('Accessible')
   })
 
-  describe('SVG Icon Accessibility', () => {
-    // Test individual icons in './lib'
+  describe.skip('SVG Icon Accessibility', () => {
```
Comment (achoboter) [inline packages/icons/test/index.js]: skipping this test?

File: packages/core/src/Avatar.js (hunk 1 above)
Comment (James300) [reply packages/core/src/Avatar.js]: Yeah that totally makes sense, thanks for bringing this up. Here is the PR for v2 Avatar https://github.com/priceline/design-system/pull/678 👍

Comment (bmansell) [reply packages/core/src/Avatar.js]: PR for Breadcrumbs v2 as well https://github.com/priceline/design-system/pull/677!

File: packages/core/src/Breadcrumbs.js (hunk 2 above)
Comment (bmansell) [reply packages/core/src/Breadcrumbs.js]: PR created! https://github.com/priceline/design-system/pull/677

**Comment by sdalonzo** [comment]
> Additional backwards compatibility with v2 is needed

@jes708 is this addresses by the PRs added by @James300 and @bmansell ?

**Comment by jes708** [comment]
> > Additional backwards compatibility with v2 is needed
> 
> @jes708 is this addresses by the PRs added by @James300 and @bmansell ?

Yep!  I'd like to merge and publish another core v2 patch for my label width fix (#680) before this PR gets merged if possible.

File: packages/core/src/Banner.js (hunk 3 above)
Comment (sdalonzo) [reply packages/core/src/Banner.js]: We should but I don't think that requires breaking change, so it need not block this. @achoboter agree?

**Comment by sdalonzo** [comment]
> > > Additional backwards compatibility with v2 is needed
> > 
> > 
> > @jes708 is this addresses by the PRs added by @James300 and @bmansell ?
> 
> Yep! I'd like to merge and publish another core v2 patch for my label width fix (#680) before this PR gets merged if possible.

Great, thanks. I already dropped an approval on #680 so that works for me!

File: packages/core/src/Radio.js (hunk 4 above)
Comment (sdalonzo) [reply packages/core/src/Radio.js]: @achoboter I pushed this change.

File: packages/core/test/__snapshots__/FormField.js.snap (hunk 6 above)
Comment (sdalonzo) [reply packages/core/test/__snapshots__/FormField.js.snap]: i have a WIP branch to fix this

File: packages/core/src/Banner.js (hunk 3 above)
Comment (sdalonzo) [reply packages/core/src/Banner.js]: cc @unscsprt

File: packages/core/test/__snapshots__/FormField.js.snap (hunk 6 above)
Comment (sdalonzo) [reply packages/core/test/__snapshots__/FormField.js.snap]: Addressed in https://github.com/priceline/design-system/pull/683

File: packages/core/test/__snapshots__/Input.js.snap (hunk 8 above)
Comment (sdalonzo) [reply packages/core/test/__snapshots__/Input.js.snap]: Addressed in https://github.com/priceline/design-system/pull/683

File: packages/core/test/__snapshots__/Select.js.snap (hunk 9 above)
Comment (sdalonzo) [reply packages/core/test/__snapshots__/Select.js.snap]: Addressed in https://github.com/priceline/design-system/pull/683

File: packages/core/test/__snapshots__/IconField.js.snap (hunk 7 above)
Comment (sdalonzo) [reply packages/core/test/__snapshots__/IconField.js.snap]: Addressed in https://github.com/priceline/design-system/pull/683

File: packages/icons/test/index.js (hunk 10 above)
Comment (sdalonzo) [reply packages/icons/test/index.js]: I'll address this today. Thanks for a good catch!

Comment (sdalonzo) [reply packages/icons/test/index.js]: Addressed in https://github.com/priceline/design-system/pull/685

**Pull request merged**